cluster/addons/metadata-proxy/gce/metadata-proxy-configmap.yaml

```diff
@@ -20,40 +20,68 @@ data:
       access_log /dev/stdout;
       server {
         listen 127.0.0.1:988;
+        # When serving 301s, don't redirect to port 988.
+        port_in_redirect off;
 
         # By default, return 403. This protects us from new API versions.
         location / {
-            return 403;
+            return 403 "This metadata API is not allowed by the metadata proxy.";
         }
 
         # Allow for REST discovery.
         location = / {
+            if ($args ~* "^(.+&)?recursive=") {
+                    return 403 "?recursive calls are not allowed by the metadata proxy.";
```

Comment:
you should mention the SECURE setting in all of these errors so people realise they are getting this error from the proxy rather than the metadata server itself, and know what to change if they need it to work.

Reply:
This isn't GKE-specific, so we can't mention the GKE feature.  The error message needs to also apply to GCE (hence 'metadata proxy').